sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -335,8 +337,12 @@ private[sql] class RocksDBStateStoreProvider
     }
 
     override def getStateStoreCheckpointInfo(): StateStoreCheckpointInfo = {
-      val checkpointInfo = rocksDB.getLatestCheckpointInfo(id.partitionId)
-      checkpointInfo
+      if (checkpointInfo == null) {
```

Comment:
I didn't see a clear reason why we don't use Scala style None, but null. Can you change to use Scala Option, or explain why?